File: app/src/main/java/com/montunosoftware/pillpopper/android/view/ProxySpinnerAdapter.java
package com.montunosoftware.pillpopper.android.view;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.montunosoftware.mymeds.R;

import org.kp.tpmg.mykpmeds.activation.model.User;

import java.util.List;

public class ProxySpinnerAdapter extends ArrayAdapter<User> {

    private List<User> spinnerData;
    LayoutInflater inflater;
    private Context context;

    public ProxySpinnerAdapter(Context context, int resourseId, List<User> usersList) {
        super(context, resourseId, usersList);
        this.context = context;
        spinnerData = usersList;
        inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public int getCount() {
        return spinnerData.size();
    }

    @Override
    public View getDropDownView(int position, View convertView, @NonNull ViewGroup parent) {
        return getCustomView(position, convertView, parent, true);
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        return getCustomView(position, convertView, parent, false);
    }

    public View getCustomView(int position, View convertView, final ViewGroup parent, boolean isDropDownView) {

        View row = convertView;

        if (row == null) {
            LayoutInflater lytInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            row = lytInflater.inflate(R.layout.user_spinner_item, parent, false);
        }
        User user = spinnerData.get(position);
        TextView txt_memberName = row.findViewById(R.id.proxy_name_textview);
        txt_memberName.setText(user.getFirstName());
        return row;
    }
}

File: app/src/main/java/com/montunosoftware/pillpopper/kotlin/history/ProxySpinnerAdapter.kt
package com.montunosoftware.pillpopper.kotlin.history

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import com.montunosoftware.mymeds.R
import org.kp.tpmg.mykpmeds.activation.model.User

class ProxySpinnerAdapter(var context: Context, val userList: List<User>) : BaseAdapter() {

    private val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val view: View
        val holder: ItemHolder
        if (null == convertView) {
            view = inflater.inflate(R.layout.history_users_spinner_listitem, parent, false)
            holder = ItemHolder(view)
            view?.tag = holder
        } else {
            view = convertView
            holder = (view.tag as ItemHolder?)!!
        }
        holder.userName?.text = userList[position].firstName
        return view
    }

    override fun getItem(position: Int): User {
        return userList[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getCount(): Int {
        return userList.size
    }

    private class ItemHolder(row: View?) {
        val userName: TextView? = row?.findViewById(R.id.history_list_proxy_name_textview)

    }
}
